examples/src/main/scala/caliban/akkahttp/ExampleApp.scala

```diff
@@ -34,21 +49,17 @@ object ExampleApp extends App with GenericSchema[Console with Clock] {
   case class Mutations(deleteCharacter: CharacterArgs => URIO[Console, Boolean])
   case class Subscriptions(characterDeleted: ZStream[Console, Nothing, String])
 
-  val interpreter = 
+  val interpreter =
     ExampleService
       .make(sampleCharacters)
-      .map(service => {
-        graphQL(
-          RootResolver(
-            Queries(
-              args => service.getCharacters(args.origin),
-              args => service.findCharacter(args.name)
-            ),
-            Mutations(args => service.deleteCharacter(args.name)),
-            Subscriptions(service.deletedEvents)
-          )
-        ).interpreter
-      })
+      .map(service => makeApi(service).interpreter)
+
+  val apiRoute: Route = AkkaHttpAdapter.makeHttpServiceM(interpreter)
+  val wsApiRoute = {
+    // when do we create subscriptions? When do we make it effectful/unsafe?
+    //  val subscriptions: UIO[Ref[Map[String, Fiber[Throwable, Unit]]]] = Ref.make(Map.empty[String, Fiber[Throwable, Unit]])
```

Comment:
`subscriptions` should be created inside the adapter, the end-user don't need to see it. A new `Ref` should be created for each WS connection.